### ISSUE TITLE
How do I handle a Tree that could be any length

### ISSUE THREAD

**Issue by zurgeg** [lark-parser/lark#546]
So I have the following grammar for my (minimal) programming language:
```
?start: value
?value: WORD
      | ifexpr
      | expr
ifexpr: "if(" [value stmt value] ")"
stmt: "is=>"
expr: add
    | subtract
    | divide
    | multiply
    | NUMBER
add:  expr "+" expr
subtract: expr "-" expr
divide: expr "/" expr
multiply: expr "*" expr


%import common.NUMBER
%import common.WORD
%ignore " "
```

How would I handle this input and solve `1+6*7` through what `p.parse('1+6*7')` returns.

**Comment by MegaIng** [comment]
Did you take a look at https://github.com/lark-parser/lark/blob/master/examples/calc.py? If yes, what exactly do you not understand?

**Comment by zurgeg** [comment]
I don't understand how I could implement the ```Transformer``` inheriting class. Also, what do the `->` symbols do?
Thanks in advance.

**Comment by MegaIng** [comment]
I would suggest you read through [json tutorial](https://github.com/lark-parser/lark/blob/master/docs/json_tutorial.md) Both your question will be answered there better than I could here.

**Comment by zurgeg** [comment]
I don't understand how to implement that class. Can you help?
Thanks in advance.

**Comment by zurgeg** [comment]
Okay so I implemented the class but it outputs `-3.0` for 1+2-1+1.
My code:
```
p = Lark(psl_grammer, parser='lalr')

prsd = p.parse('1+2-1+1')
print(prsd)

class PSTransformer(Transformer):
    def expr(self, numbers):
        tsum = 0
        addo = []
        subo = []
        mulo = []
        divo = []
        
        try:
            print(numbers)
            return float(numbers[0])
        except:
            
            print(numbers[0].children)
            if numbers[0].data == 'add':
                for i in numbers[0].children:
                    addo.append(float(i))
            elif numbers[0].data == 'subtract':
                for i in numbers[0].children:
                    subo.append(float(i))
            elif numbers[0].data == 'multiply':
                for i in numbers[0].children:
                    mulo.append(float(i))
            else:
                for i in numbers[0].children:
                    divo.append(float(i))
            for i in mulo:
                tsum = tsum * i
                print(tsum)
            for i in divo:
                tsum = tsum / i
                print(tsum)
            for i in addo:
                tsum = tsum + i
                print(tsum)
            for i in subo:
                tsum = tsum - i
                print(tsum)
            print(subo)
            print(addo)
            print(mulo)
            print(divo)
        return tsum
        
print(PSTransformer().transform(prsd))
```
Can you help? Thanks in advance.

**Comment by zurgeg** [comment]
Found the answer, here is my code and I'll explain each line:
```python
p = Lark(psl_grammer, parser='lalr')

prsd = p.parse('1+2+1+1-2*2/2')
print(prsd)

class PSTransformer(Transformer):
    def __init__(self):
        self.tsum = 0
    def expr(self, numbers):
        
        addo = []
        subo = []
        mulo = []
        divo = []
        
        try:
            print(numbers)
            rt = float(numbers[0])
            print(rt)
            #return rt
        except:
            
            if numbers[0].data == 'add':
                self.tsum = numbers[0].children[1] + numbers[0].children[0] #Embrace the communitve property!
            elif numbers[0].data == 'subtract':
                self.tsum = numbers[0].children[0] - numbers[0].children[1]
            elif numbers[0].data == 'multiply':
                self.tsum = numbers[0].children[1] * numbers[0].children[0] #Embrace the communitve property!
            if numbers[0].data == 'divide':
                self.tsum = numbers[0].children[0] / numbers[0].children[1] #Embrace the communitve property!
            rt = self.tsum
        return rt
print(PSTransformer().transform(prsd))



```
First:
```python
try:
            print(numbers)
            rt = float(numbers[0])
            print(rt)
```
That means, if it is a `NUMBER` token, then convert it to a float.

```python
 except:
            
            if numbers[0].data == 'add':
                self.tsum = numbers[0].children[1] + numbers[0].children[0] #Embrace the communitve property!
```
That means, if it is a ```Tree``` and if it is an addition problem than the value to `return` should be `a+b` or in the scenerio of the test, `1+2`,
